src/main/java/model/Student.java

```diff
@@ -0,0 +1,218 @@
+package model;
+
+import java.time.DayOfWeek;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+public class Student {
+
+    private final HashMap<LocalDateTime, AttendanceStatus> record = new HashMap<>();
+    private final String name;
+    private int absent;
+    private int attendance;
+    private int late;
+
+    public int getLate() {
+        return late;
+    }
+
+    public int getAttendance() {
+        return attendance;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public int getAbsent() {
+        return absent;
+    }
+
+    public Student(String name) {
+        this.name = name;
+    }
+
+    public void updateState(LocalDateTime localDateTime) {
+        DayOfWeek dayOfWeek = localDateTime.getDayOfWeek();
+        int day = dayOfWeek.getValue();
+        if (checkHoliday(localDateTime, day)) {
+            return;
+        }
+        AttendanceStatus attendanceStatus = AttendanceCalculatorByDay.
+                attendanceCalculator(day, LocalTime.from(localDateTime));
+
+        for (LocalDateTime localDateTime1 : record.keySet()) {
+            if (compareDayAndModify(localDateTime, localDateTime1, attendanceStatus)) {
+                return;
+            }
+        }
+
+        updateRecordForNoLocalDateTime(localDateTime, attendanceStatus);
+    }
+
+    private boolean compareDayAndModify(LocalDateTime localDateTime, LocalDateTime localDateTime1,
+                              AttendanceStatus attendanceStatus) {
+        if (compareDayIsSame(localDateTime1, localDateTime)) {
+            if (modifyByState(localDateTime, localDateTime1, attendanceStatus)) {
+                return true;
+            }
```

Comment:
boolean 반환이라면 아래와 같이 반환하는것이 깔끔할 것 같아요!

```suggestion
        return compareDayIsSame(localDateTime1, localDateTime) && modifyByState(localDateTime, localDateTime1, attendanceStatus);
```

Reply:
반영하여 수정하겠습니다!!